crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -729,6 +1030,103 @@ mod tests {
+        let context = test_cluster.wallet;
+        let config = FaucetConfig {
+            ttl_expiration: 10,
+            ..Default::default()
+        };
+        let prom_registry = Registry::new();
+        let tmp = tempfile::tempdir().unwrap();
+        let faucet = SimpleFaucet::new(
+            context,
+            &prom_registry,
+            &tmp.path().join("faucet.wal"),
+            config,
+        )
+        .await
+        .unwrap();
+
+        let amounts = &vec![1; 1];
+        // Create a vector containing five randomly generated addresses
+        let target_addresses: Vec<SuiAddress> = (0..5)
+            .map(|_| SuiAddress::random_for_testing_only())
+            .collect();
+
+        let response = futures::future::join_all(
+            target_addresses
+                .iter()
+                .map(|address| faucet.batch_send(Uuid::new_v4(), *address, amounts)),
+        )
+        .await
+        .into_iter()
+        .map(|res| res.unwrap())
+        .collect::<Vec<BatchFaucetReceipt>>();
+
+        // Assert that all of these return in progress
+        let status_results = futures::future::join_all(
+            response
+                .clone()
+                .iter()
+                .map(|task| faucet.get_batch_send_status(Uuid::parse_str(&task.task).unwrap())),
+        )
+        .await
+        .into_iter()
+        .map(|res| res.unwrap())
+        .collect::<Vec<BatchSendStatus>>();
+
+        for status in status_results {
+            assert_eq!(status.status, BatchSendStatusType::INPROGRESS);
+        }
+
+        // Transfer the gases
+        faucet
+            .batch_transfer_gases()
+            .await
+            .expect("Batch transfer failed.");
+
+        // Assert that all of these are SUCCEEDED
+        let status_results = futures::future::join_all(
+            response
+                .clone()
+                .iter()
+                .map(|task| faucet.get_batch_send_status(Uuid::parse_str(&task.task).unwrap())),
+        )
+        .await
+        .into_iter()
+        .map(|res| res.unwrap())
+        .collect::<Vec<BatchSendStatus>>();
+
+        for status in status_results {
+            assert_eq!(status.status, BatchSendStatusType::SUCCEEDED);
+        }
+
+        // Check that TTL cache expires
+        tokio::time::sleep(Duration::from_secs(10)).await;
+        let status_results = futures::future::join_all(
+            response
+                .clone()
+                .iter()
+                .map(|task| faucet.get_batch_send_status(Uuid::parse_str(&task.task).unwrap())),
+        )
+        .await;
+
+        let all_errors =
+            status_results.iter().all(
+                |status_result| {
+                    if status_result.is_err() {
+                        true
+                    } else {
+                        false
+                    }
+                },
+            );
```

Comment:
This should work:

```suggestion
        let all_errors = status_results.iter().all(Result::is_err);
```

Failing that, just return `is_err` directly (i.e. no need for the `if`).